storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -433,4 +461,49 @@ private void initConfigs() {
     public Double getTopologyWorkerMaxHeapSize() {
         return this.topologyWorkerMaxHeapSize;
     }
+
+    /**
+     * Get the user that submitted this topology
+     */
+    public String getTopologySubmitter() {
+        String user = (String) this.topologyConf.get(Config.TOPOLOGY_SUBMITTER_USER);
+        if (user == null || user.equals("")) {
+            LOG.debug("Topology {} submitted by anonymous user", this.getName());
+            user = "anonymous";
```

Comment:
Now that we merged in the blob store code the user is never going to be null or "".  Because it was causing issues elsewhere in the code.  We should either throw an exception, or default to `System.getProperty("user.name")` like nimbus does.